core/sql/generator/GenPreCode.cpp

```diff
@@ -11274,6 +11274,485 @@ short HbaseAccess::extractHbaseFilterPreds(Generator * generator,
+	       (NOT hasColReference(ie->child(0))))
+	{
+      foundBinary = TRUE;
+      flipOp = TRUE;
+	  colVID = ie->child(1)->getValueId();
+	  valueVID = ie->child(0)->getValueId();
+	}
+      else if ((ie->child(0)->getOperatorType() == ITM_REFERENCE) &&
+	       (NOT hasColReference(ie->child(1))))
+	{
+      foundBinary = TRUE;
+	  colVID = ie->child(0)->getValueId();
+	  valueVID = ie->child(1)->getValueId();
+	}
+      else if ((ie->child(1)->getOperatorType() == ITM_REFERENCE) &&
+	       (NOT hasColReference(ie->child(0))))
+	{
+      foundBinary = TRUE;
+      flipOp = TRUE;
+	  colVID = ie->child(1)->getValueId();
+	  valueVID = ie->child(0)->getValueId();
+	}
+      else if ((ie->child(0)->getOperatorType() == ITM_HBASE_COLUMN_LOOKUP) &&
+	       (NOT hasColReference(ie->child(1))))
+	{
+	  HbaseColumnLookup * hcl = (HbaseColumnLookup*)ie->child(0)->castToItemExpr();
+	  if (hcl->getValueId().getType().getTypeQualifier() == NA_CHARACTER_TYPE)
+	    {
+	      hbaseLookupPred = TRUE;
+
+	      ItemExpr * newCV = new(generator->wHeap()) ConstValue(hcl->hbaseCol());
+	      newCV = newCV->bindNode(generator->getBindWA());
+	      newCV = newCV->preCodeGen(generator);
+
+	      foundBinary = TRUE;
+	      colVID = newCV->getValueId();
+	      valueVID = ie->child(1)->getValueId();
+	    }
+	}
+      else if ((ie->child(1)->getOperatorType() == ITM_HBASE_COLUMN_LOOKUP) &&
+	       (NOT hasColReference(ie->child(0))))
+	{
+	  HbaseColumnLookup * hcl = (HbaseColumnLookup*)ie->child(1)->castToItemExpr();
+	  if (hcl->getValueId().getType().getTypeQualifier() == NA_CHARACTER_TYPE)
+	    {
+	      hbaseLookupPred = TRUE;
+
+	      ItemExpr * newCV = new(generator->wHeap()) ConstValue(hcl->hbaseCol());
+	      newCV = newCV->bindNode(generator->getBindWA());
+	      newCV = newCV->preCodeGen(generator);
+
+	      foundBinary = TRUE;
+	      flipOp = TRUE;
+	      colVID = newCV->getValueId();
+	      valueVID = ie->child(0)->getValueId();
+	    }
+	}
+    }//end binary operators
+  else if (ie && ((ie->getOperatorType() == ITM_IS_NULL)||(ie->getOperatorType() == ITM_IS_NOT_NULL))){//check for unary operators
+	  ItemExpr * child0 = ie->child(0)->castToItemExpr();
+	  if ((ie->child(0)->getOperatorType() == ITM_BASECOLUMN) ||
+		  (ie->child(0)->getOperatorType() == ITM_INDEXCOLUMN)||
+		  (ie->child(0)->getOperatorType() == ITM_REFERENCE)){
+		  foundUnary = TRUE;
+		  colVID = ie->child(0)->getValueId();
+		  valueVID = NULL_VALUE_ID;
+	  }
+
+  }//end unary operators
+
+  //check if found columns belong to table being scanned (so is not an input to the scan node)
+  if (foundBinary || foundUnary){
+	ValueId dummyValueId;
+	if (getGroupAttr()->getCharacteristicInputs().referencesTheGivenValue(colVID,dummyValueId)){
+		foundBinary=FALSE;
+		foundUnary=FALSE;
+	}
+  }
+  //check if not an added column with default non null
+  if ((foundBinary || foundUnary)&& (NOT hbaseLookupPred)){
+	  NAColumn * nac;
```

Comment:
Should initialize to null. Otherwise the default case will be taken at line 11425, and God knows what will happen when you reach line 11428 (but in release builds its not likely to be pretty).